### DISCUSSION TITLE
GSoC 2022 interested in topic: "JaCoCo support"

### DISCUSSION THREAD

**Discussion by ZSmallX** [robolectric/robolectric#7128] [General]
I'm interested in robolectric JaCoCo support. I'm working on Android Testing Tools in my job and had met similar problem before. And I had some investigations in this problem as well. It's a hard and tricky problem for many years in many comunities, where JaCoCo, Mock frameworks(PowerMock etc.) and Gradle get involved.

Here's part of my previous work and solution using JaCoCo offline instrumentation for CI automation. It may provide some context and inspiration on IDE problem.
https://github.com/robolectric/robolectric/issues/3023#issuecomment-1063112712 and https://github.com/robolectric/robolectric/issues/3023#issuecomment-1064174088.

Since I don't actually get much into java agents conflict details, I might have no real good solution for this.

From my understanding, there probably needs to be an “orchestrator” on the build side or IDE side to make those modules work well together. Or I'm not sure if it can be done within Rorolectric.

 Hope we can have more discussions and work on it.

**Comment by utzcoz** [comment]
@hoisie @ZSmallX As we discussed before, we should set this project's expected output properly based on current investigation. It has very large possibility related to Android Studio/Gradle, and it's difficult for us to fix related problems for them. I know GSoC needs us to set expected output for program to evaluate project's final result. The next few weeks(about two weeks) we should discuss more about it to find the things we can do for this project. IMO this project is very important for Robolectric, because many folks have encountered similar problem when running Robolectric tests from Android Studio.
